newIDE/app/src/SceneEditor/index.js

```diff
@@ -282,6 +282,13 @@ export default class SceneEditor extends React.Component<Props, State> {
     this.setState({ scenePropertiesDialogOpen: open });
   };
 
+  saveUiSettings = () => {
```

Comment:
This should not be in this file ;) Just noticed it.

I'm even wondering if this is working - have you tried?. Because saveUiSettings will never be found. In fact, `saveUiSettings` should be added to `newIDE/app/src/MainFrame/Editors/SceneEditor.js` and `newIDE/app/src/MainFrame/Editors/ExternalLayoutEditor.js` (as I mentionned in a previous comment, this has to work also for external layout, otherwise we'll have a half baked feature).

This is because SceneEditor here is the component that is used to edit a scene or an external layout. All the tabs, containing the editors, are in `MainFrame/Editors`.

It's true that the name can be confusing, as both are called SceneEditor.

Reply:
It does work :) before writing any save logic I even got it to print out a message  from there.
I tested saving and loading of grid settings too.

Can you test it please

Reply:
I can move it if you want to, but I think moving it might actually break it

Reply:
My bad, I didn't see the extra `.editor` in EditorTabsHandler.js. This might work but also make things fragile to change - if I decide to rename this variable (I should have put an underscore: _editor to make it clear) later, it will break this feature.
Actually, editor can even be null if for some reason a tab is opened but refering to an improper layout.

`editor` is a "ref", it's a way to get a reference to a React component. We should avoid using them outside of the component where they are declared.

Reply:
Ok this is refactored now and still works. Can also save settings of external layouts

Reply:
need to add this.editor to the check before unserializing, since the flow will fail to pass otherwise